docs/dev_guide/router_rest_spec.md

```diff
@@ -0,0 +1,261 @@
+---
+layout: default
+title: Router REST Spec
+parent: Developer Guides
+permalink: /docs/dev_guide/router_rest_spec
+---
+# Router REST Spec
+This page describes how the router's REST endpoints work for reading data out of Venice with the thin client. Currently,
+there exists a reference implementation of the thin client in Java, but if clients in additional languages are to be 
+built, they could reference this spec to do so.
+
+Note that although this part of the architecture is RESTful, Venice in general has a bias for performance, so the
+implementation may not be fully up to the standards of a dogmatic RESTafarian. For example, because of URL length
+limitations, batch gets are implemented with a POST rather than a GET, even though it carries no side effects.
+
+The following sections describe the endpoints in the order they are invoked by the client during initialization, in
+order to prime its internal state.
+
+## Cluster Discovery Endpoint
+The first endpoint the client invokes is to discover which cluster the store of interest belongs to:
+
+```
+https://<host>:<port>/discover_cluster/<store_name>
+```
+
+This will return a JSON response in the following format:
+
+```json
+{
+  "cluster": "cluster_name",
+  "name": "store_name",
+  "error": null,
+  "errorType": null,
+  "d2Service": "D2_SERVICE_NAME_62513b39957e_af278fc8",
+  "serverD2Service": "SERVER_D2_SERVICE_NAME_62513b3d13a5_4dc8aa80",
+  "zkAddress": "localhost:63359",
+  "kafkaBootstrapServers": "localhost:1234",
```

Comment:
Also... @avedmala has since then added a field to `D2ServiceDiscoveryResponse` so 🙈 😅

Reply:
Yeah, I told him we could do that since we had marked all legacy clients as deprecated. I can make this change in the ControllerClient and ControllerAPI PR: https://github.com/linkedin/venice/pull/427